src/Compilers/CSharp/Portable/BoundTree/BoundPatternSwitchStatement.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class BoundPatternSwitchStatement
+    {
+        public BoundDecisionDag DecisionDag { get; }
+
+        public BoundPatternSwitchStatement(SyntaxNode syntax, BoundExpression expression, ImmutableArray<LocalSymbol> innerLocals, ImmutableArray<LocalFunctionSymbol> innerLocalFunctions, ImmutableArray<BoundPatternSwitchSection> switchSections, BoundPatternSwitchLabel defaultLabel, GeneratedLabelSymbol breakLabel, BoundDecisionDag decisionDag, bool hasErrors = false)
+            : base(BoundKind.PatternSwitchStatement, syntax, hasErrors || expression.HasErrors() || switchSections.HasErrors() || defaultLabel.HasErrors() || decisionDag.HasErrors())
+        {
+
+            Debug.Assert(expression != null, "Field 'expression' cannot be null (use Null=\"allow\" in BoundNodes.xml to remove this check)");
+            Debug.Assert(!innerLocals.IsDefault, "Field 'innerLocals' cannot be null (use Null=\"allow\" in BoundNodes.xml to remove this check)");
+            Debug.Assert(!innerLocalFunctions.IsDefault, "Field 'innerLocalFunctions' cannot be null (use Null=\"allow\" in BoundNodes.xml to remove this check)");
+            Debug.Assert(!switchSections.IsDefault, "Field 'switchSections' cannot be null (use Null=\"allow\" in BoundNodes.xml to remove this check)");
+            Debug.Assert(breakLabel != null, "Field 'breakLabel' cannot be null (use Null=\"allow\" in BoundNodes.xml to remove this check)");
+
+            this.Expression = expression;
+            this.InnerLocals = innerLocals;
+            this.InnerLocalFunctions = innerLocalFunctions;
+            this.SwitchSections = switchSections;
+            this.DefaultLabel = defaultLabel;
+            this.BreakLabel = breakLabel;
+            this.DecisionDag = decisionDag;
+        }
```

Comment:
>} [](start = 8, length = 1)

Could this constructor call the generated constructor and simply set `DecisionDag`? Or could `DecisionDag` be added to the definition in `BoundNodes.xml`?

Reply:
Yes, could call the generated constructor.

`DecisionDag` cannot be added to the definition in `BoundNodes.xml` because the generated visitors would take exponential time due to revisiting dag nodes with more than one parent.